src/pages/signin/SAMLEnabledForm.js

```diff
@@ -0,0 +1,109 @@
+import * as Session from '../../libs/actions/Session';
+import ChangeExpensifyLoginLink from './ChangeExpensifyLoginLink';
+import Terms from './Terms';
+import CONST from '../../CONST';
+import ROUTES from '../../ROUTES';
+import Navigation from '../../libs/Navigation/Navigation';
+
+const propTypes = {
+    /* Onyx Props */
+
+    /** The credentials of the logged in person */
+    credentials: PropTypes.shape({
+        /** The email/phone the user logged in with */
+        login: PropTypes.string,
+    }),
+
+    /** The details about the account that the user is signing in with */
+    account: PropTypes.shape({
+        /** Whether or not a sign on form is loading (being submitted) */
+        loading: PropTypes.bool,
+    }),
+
+    /** Information about the network */
+    network: networkPropTypes.isRequired,
+
+    /** Function to change whether the user is using SAML or magic codes to log in */
+    setIsUsingSAMLLogin: PropTypes.func.isRequired,
+
+    ...withLocalizePropTypes,
+
+    ...windowDimensionsPropTypes,
+};
+
+const defaultProps = {
+    credentials: {},
+    account: {},
+};
+
+function SAMLEnabledForm (props) {
+    return (
+        <>
+            <View>
+                <Text style={[styles.loginHeroBody, styles.mb5, styles.textNormal, !props.isSmallScreenWidth ? styles.textAlignLeft : {}]}>
+                    {props.translate('samlSignIn.welcomeSAMLEnabled')}
+                </Text>
+                <Button
+                    isDisabled={props.network.isOffline}
+                    success
+                    style={[styles.mv3]}
+                    text={props.translate('samlSignIn.useSingleSignOn')}
+                    isLoading={props.account.isLoading}
+                    onPress={() => {
+                        Navigation.navigate(ROUTES.SAML_SIGN_IN);
+                    }}
+                />
+
+                <View style={[styles.mt5]}>
+                    <Text style={[styles.loginHeroBody, styles.mb5, styles.textNormal, !props.isSmallScreenWidth ? styles.textAlignLeft : {}]}>
+                        {props.translate('samlSignIn.orContinueWithMagicCode')}
+                    </Text>
+                </View>
+
+                <Button
+                    isDisabled={props.network.isOffline}
+                    style={[styles.mv3]}
+                    text={props.translate('samlSignIn.useMagicCode')}
+                    isLoading={
+                        props.account.isLoading && props.account.loadingForm === (props.account.requiresTwoFactorAuth ? CONST.FORMS.VALIDATE_TFA_CODE_FORM : CONST.FORMS.VALIDATE_CODE_FORM)
+                    }
+                    onPress={() => {
+                        Session.resendValidateCode(props.credentials.login);
+                        props.setIsUsingSAMLLogin(false);
+                    }}
+                />
+                <ChangeExpensifyLoginLink onPress={() => Session.clearSignInData()} />
```

Comment:
Also, set `isUsingSAMLLogin` to false here?

Reply:
We can, but in this case they haven't actually made a decision on which method to use, they're just re-starting the sign in process 🤔

Reply:
Yes, so they will use a new login and `isUsingSAMLLogin` will be populated again?
I don't think we'll use the value if we don't clear it. So better to clear it?

Reply:
Yeah, I mean we basically just want to clear the state whenever the `LoginPage` is shown but setting it to `false` here might not be correct (i.e. if the new login has SAML enabled as well)